File: Codes/0342-power-of-four_20250815.go
/*
 * @Author: LetMeFly
 * @Date: 2025-08-15 18:29:00
 * @LastEditors: LetMeFly.xyz
 * @LastEditTime: 2025-08-15 23:54:26
 */
package main

func isPowerOfFour(n int) bool {
    return n > 0 && n & (n - 1) == 0 && n % 3 == 1
}

File: Codes/0342-power-of-four_20250815.rs
/*
 * @Author: LetMeFly
 * @Date: 2025-08-15 18:29:00
 * @LastEditors: LetMeFly.xyz
 * @LastEditTime: 2025-08-15 23:57:22
 */
impl Solution {
    pub fn is_power_of_four(n: i32) -> bool {
        n > 0 && n & (n - 1) == 0 && n % 3 == 1
    }
}